### DISCUSSION TITLE
German translations of django CMS elements

### DISCUSSION THREAD

**Discussion by fsbraun** [django-cms/django-cms#7631] [General]
With the upcoming django CMS v4 we are also updating all translations. It turns out that so far, there have not been consistent translations for "Placeholder" and "Plugin" in the French locale. French translations are moving to consistently using "_Emplacement_" and "_Bloc_" (_bloc_ has already been used in French in v3, at least at some places).

The German translations in v3 for Placeholder and Plugin are "_Platzhalter_" and "_Plugin_", respecitvely. 

**I propose to change them to "_Platzhalter_" and "_Block_", respectively**. This would simplify the work for editors. They would deal with a "_Container-Block_", a "_Text-Block_", etc. It will be consistent with the French translation. Program code will not change, i.e. the "_Text-Block_" will still be coded as a subclass of `CMSPluginBase`.

Please let me know if you feel this is an improvement for the German-speaking user base of django CMS. I will include the new translations for testing in django CMS 4.1.0rc4 with a clear understanding that they might be reverted as a result of this discussion.

**Comment by macolo** [comment]
I am in favour of this translation-only change. I would also consider changing this in the English language (default language) to something like "block" or "element" (preferably something that is consistent with the other language translations).

It would also be interesting to check how other modern CMS (sanity.io, storyblok.com and contentful.com) are dealing with the naming.

**Reply by fsbraun** [reply]
@rolandf created an overview and template for en and fr here: https://docs.google.com/spreadsheets/d/1mThjzET8P6CQciLFvH5NZB4x1ksWlDewNrZoP92wFF0/edit

**Reply by raffael-mnhn** [reply]
I also like the idea of moving away from the name "plugin" to something more understandable to non-programmers. I like the idea of "block".

**Comment by jrief** [comment]
Imo, the wording for Placeholder=Platzhalter makes perfectly sense. Everybody immediately understand what this means.

I'm struggling a bit with plugin. This term is a) too generic and b) usually used for something else. Therefore renaming "Plugin" to something like "Content-Element", "Content-Component" or similar would be really beneficial. Drupal for instance uses "Content-Type" whose term is already used by the Django itself.

**Reply by fsbraun** [reply]
As a side-note: I was also considering "Baustein" (think Lego), two syllables just like plugin, but "Text-Baustein" already has a different meaning and "Text-Block" made more sense to me. What do you think?

"Komponente" already has 4 syllables. It should not be cumbersome to speak about django CMS...

**Reply by raffael-mnhn** [reply]
Why not just "Block" in German as well?

**Reply by jrief** [reply]
because a "Block" usually refers to something rectangular. In djangocms-cascade I have Plugins which are not visible at all, just wrappers, hence they can by no means be considered as "Blocks".

**Reply by fsbraun** [reply]
"Baustein" would translate to "building block". Maybe that's sufficiently close?

**Reply by raffael-mnhn** [reply]
Well, plug-ins usually plug into something. That's pretty far fetched as well, although I see the reasoning behind it. I think it's more important to find a term that immediately clicks (which I think "Block" is) than to find the perfect analogy. On top of that, most plug-ins in django-cms that I've come along have been rectangular. I can certainly say that the first time I came in contact with django-cms I was slightly confused by the "plugin" term. Plug-ins have usually evoked the notion of a singular extension to me, ie. something that is not instantiated multiple times.

On top of the term "Block" being well-known for most German speakers, it would also be nice to be aligned with the English term (if that's going to happen). I also find "Baustein" pretty good. It immediately evokes constructing something larger (a Page) from it.

Just my 2 cents.

**Reply by jrief** [reply]
And as we all know, there are two hard things in computer science: cache invalidation and naming things. "Plugin" wasn't a good choice.

**Comment by corentinbettiol** [comment]
> With the upcoming django CMS v4 we are also updating all translations. It turns out that so far, there have not been consistent translations for "Placeholder" and "Plugin" in the French locale.

Since slack will hide the messages, here's a [screenshot](https://github.com/django-cms/django-cms/assets/45763865/4f12434b-5f4a-4ca1-94fd-692f8c9d78d3) of the discussion on the djangocms slack server, for posterity.
